modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -315,34 +319,34 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
     {
         BaseConvolutionLayerImpl::finalize(inputs_arr, outputs_arr);
 
-        CV_Assert(!blobs.empty());
-        const int outCn = blobs[0].size[0];
+        std::vector<Mat> inputs;
+        inputs_arr.getMatVector(inputs);
         // prepare weightsMat where each row is aligned and has enough zero padding on the right to
         // use vectorized (i.e. with intrinsics) loops without tail processing
-        Mat wm = blobs[0].reshape(1, outCn);
+        Mat wm = blobs.empty() ? inputs[1].reshape(1, numOutput) : blobs[0].reshape(1, numOutput);
```

Comment:
Is `numOutput` always correct?

Reply:
`numOutput` is a required parameter, so I think it is always correct